chore(dev-infra): cleanup package dependencies for shared package

devversion opened the pull request:
As per our discussion in the dev-infra sync meeting, we don't want to
have all dependencies show up as peer dependencies. Instead, we only
want to have larger dependencies such as `typescript` or buildifier as
peer dependencies. Tslib is also included for the sake of it being generally
a peer dependency of all Angular framework packages.

The rationale is that Yarn is smart enough to collapse packages if all
satisfy a given range. This means that we don't necessarily need to have
all dependencies as peer dependencies. The initial idea was to keep
all dependencies as peer dependencies so that we have control over
duplication of packages as downloading multiple packages w/ different
versions impacts local dev, CI and caches.

At the same time though, we don't want to bother with setting up peer
dependencies all the time. Not every consumer of the shared dev-infra
package would like to manually specify `yaml` or `multimatch` etc. in the
project `package.json`. Hence we decided to go with a hybrid approach
where only more impactful dependencies are peer dependencies, and
other smaller ones can be standard dependencies that are usually collapsed
by Yarn anyway.

Also this commit removes tslib from build targets that don't rely on it.

josephperrott commented on dev-infra/tmpl-package.json:
```diff
@@ -18,9 +17,12 @@
+    "yargs": "<from-root>"
+  },
+  "peerDependencies": {
+    "@bazel/buildifier": "<from-root>",
```
We should also include `clang-format` in peer deps as a match for our handling of `@bazel/buildifier`

devversion replied:
Good point. Done!

josephperrott reviewed (approved):
LGTM, one small comment

devversion committed:
build(dev-infra): cleanup package dependencies for shared package

As per our discussion in the dev-infra sync meeting, we don't want
to have all dependencies show up as peer dependencies. Instead, we
only want to have larger dependencies such as `typescript` or buildifier
as peer dependencies. Tslib is also included for the sake of it being
generally a peer dependency of all Angular framework packages.

The rationale is that Yarn is smart enough to collapse packages
if all satisfy a given range. This means that we don't necessarily
need to have all dependencies as peer dependencies. The initial
idea was to keep all dependencies as peer dependencies so that
we have control over duplication of packages as downloading multiple
packages w/ different versions impacts local dev, CI and caches.

At the same time though, we don't want to bother with setting
up peer dependencies all the time. Not every consumer of the
shared dev-infra package would like to manually specify `yaml`
or `multimatch` etc. in the project `package.json`. Hence we
decided to go with a hybrid approach where only more impactful
dependencies are peer dependencies, and other smaller ones can
be standard depdencies that are usually collapsed by Yarn anyway.

Also this commit removes tslib from build targets that don't
rely on it.
